bin_sort/BinSort.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Random;
+
+class Node {
+	public double data;
+	public Node next;
+
+	public Node(double data) {
+		this.data = data;
+		next = null;
+	}
+}
+
+// Time Complexity O(n)-> avg case and O(n^2)-> Worst Case
+// constraints:- 0<=A[i]<1
+public class BinSort {
+	public static void binSort(Node[] b, double[] a) {
+		for (int i = 0; i < a.length; i++) {
+			int ins = (int) (a[i] * 10);

```

Comment:
Here 10 is used to convert that fraction```a[i]``` into integer i.e fraction like 0.123 converted into 1 so it can be used as index. I believe this 10 should remain as it is . What's your opinion @aviaryan ?